tests/CMakeLists.txt

```diff
@@ -8,27 +8,25 @@ enable_testing()
 ################################################################################
 # C tests and tools
 
+set(COVERAGE_TARGETS)
+# Macro to register a test for coverage. The first argument is the target name.
+# Other arguments, like data path, can be added.
+macro(register_test_for_coverage TEST_NAME)
+    add_custom_target(
+        ${TEST_NAME}_coverage
+        COMMAND ${CMAKE_COMMAND} -E env "LLVM_PROFILE_FILE=${CMAKE_CURRENT_BINARY_DIR}/${TEST_NAME}.profraw"
+                $<TARGET_FILE:${TEST_NAME}> ${ARGN}
+    )
+    list(APPEND COVERAGE_TARGETS ${TEST_NAME})
+endmacro()
```

Comment:
Note: This only applies to clang. This should still build because it only adds an unused environment target.

Reply:
Indeed. I now disable `AVIF_ENABLE_COVERAGE` when it is not available.
I also do not define those targets if disabled, to lighten the list of targets.